linkerd/app/gateway/src/server.rs

```diff
@@ -75,20 +76,49 @@ impl Gateway {
             .into_inner();
 
         let discover = {
-            use profiles::GetProfile;
-
-            // Apply the gateway's allowlist to the profile discovery service.
-            let allowlist = self.config.allow_discovery.clone().into();
-            let mut profiles = profiles::WithAllowlist::new(profiles, allowlist);
-            self.outbound
-                .with_stack(protocol)
-                .push_discover(svc::mk(move |GatewayAddr(addr)| {
-                    profiles.get_profile(profiles::LookupAddr(addr.into()))
-                }))
-                .into_stack()
+            use futures::future;
+
+            let allowlist = self.config.allow_discovery.clone();
+            svc::mk(move |GatewayAddr(addr)| {
```

Comment:
I'd generally prefer to split this into a function like
```rust
    fn discovery(
        &self,
        profiles: impl profiles::GetProfile,
        policies: impl outbound::policy::GetPolicy,
    ) -> impl svc::Service<
        GatewayAddr,
        Response = (
            Option<profiles::Receiver>,
            watch::Receiver<outbound::policy::ClientPolicy>,
        ),
        Error = Error,
        Future = impl Send + Unpin,
    > + Clone {
        let allowlist = self.config.allow_discovery.clone();
        svc::mk(move |GatewayAddr(addr)| {
          ...
        })
    }
```

So that we can more clearly delineate "discovery stuff" from stack building stuff

Reply:
oh, okay, i actually did that initially but then undid it because i wasn't sure if it was worth it. can put it back like that.